src/topology/separation.lean

```diff
@@ -937,6 +937,18 @@ lemma continuous.ext_on [t2_space α] {s : set β} (hs : dense s) {f g : β →
   f = g :=
 funext $ λ x, h.closure hf hg (hs x)
 
+/-- If `f x = g x` for all `x ∈ s` and `f`, `g` are continuous on `t`, `s ⊆ t ⊆ closure s`, then
+`f x = g x` for all `x ∈ t`. -/
+lemma set.eq_on.of_subset_closure [t2_space α] {s t : set β} {f g : β → α} (h : eq_on f g s)
```

Comment:
Can you mention in the docstring of `set.eq_on.closure` that `set.eq_on.of_subset_closure` is a generalization?